web/src/pages.rs

```diff
@@ -146,3 +151,77 @@ pub async fn my_articles(
     })
     .into_response())
 }
+
+#[axum::debug_handler(state = AppState)]
+pub async fn my_categories(
+    t: Template,
+    State(state): State<AppState>,
+    user: User,
+) -> WebResult<Response> {
+    info!("Fetching categories for user_id: {}", user.user_id);
+
+    let categories = sqlx::query!(
+        "SELECT DISTINCT c.* 
+     FROM category c
+     JOIN categorymarkdown cm ON c.category_id = cm.category_id
+     JOIN markdown m ON cm.markdown_id = m.markdown_id
+     JOIN pagesnapshot ps ON m.page_snapshot_id = ps.page_snapshot_id
+     JOIN pages p ON ps.page_id = p.page_id
+     WHERE p.user_id = $1",
+        user.user_id
+    )
+    .fetch_all(&state.db)
+    .await?;
+
+    Ok(t.render(maud::html! {
+        h1 { "My Categories" }
+        ul {
+            @for category in categories {
+                @let category_name = category.category.unwrap_or("No category".to_string());
+                li {
+                    a href=(format!("/categories/{}", category.category_id)) { (category_name) }
+                }
+            }
+        }
+    })
+    .into_response())
+}
+
+#[axum::debug_handler(state = AppState)]
+pub async fn articles_by_category(
+    t: Template,
+    Path(category_id): Path<Uuid>,
+    State(state): State<AppState>,
+    user: User,
+) -> WebResult<Response> {
+    info!("Fetching articles for category_id: {}", category_id);
+
+    let articles = sqlx::query!(
+        r#"
+        SELECT m.summary, p.url
+        FROM category c
+        JOIN categorymarkdown cm ON c.category_id = cm.category_id
+        JOIN markdown m ON cm.markdown_id = m.markdown_id
+        JOIN pagesnapshot ps ON m.page_snapshot_id = ps.page_snapshot_id
+        JOIN pages p ON ps.page_id = p.page_id
+        WHERE c.category_id = $1 AND p.user_id = $2
+        "#,
+        category_id,
+        user.user_id
+    )
+    .fetch_all(&state.db)
+    .await?;
+
+    Ok(t.render(maud::html! {
+        h1 { "Articles in Category" }
+        ul {
+            @for article in articles {
+                li {
+                    a href=(article.url) { (article.url) }
```

Comment:
Can we link to the `articles/:id` here instead of directly to the actual article URL?

That way you can click on the link to get our 'detail' pages!